### PR TITLE
R package fixes & improvements

### PR DESCRIPTION
**Pull request by sllynn**
Updates to the boilerplate parts of sparklyr bindings and addition of some testthat tests to both sparklyr and sparkr bindings.

### CODE REVIEW

File: R/build_r_package.R [hunk 1]
```diff
@@ -1,11 +1,10 @@
-spark_location <- "/usr/spark-download/unzipped/spark-3.2.1-bin-hadoop2.7"
+spark_location <- "/usr/spark-download/unzipped/spark-3.3.2-bin-hadoop3"
```
Comment (milos-colic) [inline R/build_r_package.R:1]: @sllynn should this also refer to spark.matrix?

File: R/generate_docs.R [hunk 2]
```diff
@@ -1,4 +1,4 @@
-spark_location <- "/usr/spark-download/unzipped/spark-3.2.1-bin-hadoop2.7"
+spark_location <- "/usr/spark-download/unzipped/spark-3.3.2-bin-hadoop3"
```
Comment (milos-colic) [inline R/generate_docs.R:1]: @sllynn should this refer to spark.matrix?

**Review by milos-colic — COMMENTED** [review]
Just 2 comments, looks good to me in general

File: R/generate_docs.R (hunk 2 above)
Comment (sllynn) [reply R/generate_docs.R:1]: I can probably put it in an environment variable somewhere 🤔

**Review by milos-colic — COMMENTED** [review]
LGTM

**Review by milos-colic — APPROVED** [review]

**Pull request merged**